### PR TITLE
Upgraded react-datatable SPFx to version 1.20.0

### PR DESCRIPTION
**Pull request by mohammadamer**
- [ ] New sample
- [x] Bug fix/update #5248
- [ ] Related issues: fixes #X, partially #Y, mentioned in #Z

## What's in this Pull Request?
Upgraded react-datatable SPFx to version 1.20.0
Fixed a bug when exporting to PDF

## Node Version
20.16.0

Node version used:
20.16.0

## Checklist

- [x] My pull request affects only ONE sample.
- [ ] My sample builds without any warnings
- [x] I have updated the `README.md` file's **Version history**. For new samples, created a new `README.md` file matching [this template](templates/README-template.md)
- [x] My `README.md` has at least one static high-resolution screenshot (i.e. not a GIF) located in the `assets` folder.
- [x] My `README.md` contains complete setup instructions, including pre-requisites and permissions required
- [x] My solution includes a `.nvmrc` file indicating the version of Node.js

### CODE REVIEW

**Review by Adam-it — CHANGES_REQUESTED** [review]
@mohammadamer awesome work 👏👏👏👏 
I managed to build the solution and play with it. 
I noticed some tiny details we could sort out before we merge. Please do give them a double check 🙏

File: samples/react-datatable/.nvmrc
Comment (mohammadamer) [reply samples/react-datatable/.nvmrc:1]: Well, As per the upgrade file generated by the SPFx toolkit, I did use version 20.16.0
But if it works with 18 then I can also change it. What exactly 18 version to replace?

`{
  "@microsoft/generator-sharepoint": {
    "nodeVersion": "20.16.0"
  }
}`

File: samples/react-datatable/.yo-rc.json
Comment (mohammadamer) [reply samples/react-datatable/.yo-rc.json:4]: Then Please tell me what exactly the 18 node version to include.

File: samples/react-datatable/README.md
Comment (mohammadamer) [reply samples/react-datatable/README.md:1]: I would suggest to add check in the PR templates for these 2 updates
1. assets/sample.json
2. devcontainer.json

**Comment by mohammadamer** [comment]
@Adam-it 
All comments are resolved :)

File: samples/react-datatable/.yo-rc.json
Comment (Adam-it) [reply samples/react-datatable/.yo-rc.json]: SPFx 1.20 works with any version of Node v18 as we may find in the MS docs
image

File: samples/react-datatable/.nvmrc
Comment (Adam-it) [reply samples/react-datatable/.nvmrc]: Sorry for the late reply 🙏
We may use any version of Node v18 for SPFx 1.20 from what I am aware so I would go w latest.
as for the snippet you provided. May I kindly ask you about a bit more context. When running which action/command you get this result? is it part of some bigger output?

File: samples/react-datatable/README.md
Comment (Adam-it) [reply samples/react-datatable/README.md:1]: @mohammadamer this is a good suggestion indeed. I will tag @hugoabernier as he is the maintainer of this repo to check this suggestion. I may only add that sample updates are maybe not something done on regular bases so I am not entirely sure this should be part of PR template 🤔

**Review by Adam-it — APPROVED** [review]
Looks great now 👏👏👏

**Pull request merged**
